varnish/tests/test_varnish.py

```diff
@@ -10,28 +11,35 @@
 
 
 def test_check(aggregator, check, instance):
-    check.check(instance)
+    _check_repeat(check, instance)
     for mname in common.COMMON_METRICS:
-        aggregator.assert_metric(mname, count=1, tags=['cluster:webs', 'varnish_name:default'])
+        aggregator.assert_metric(mname, at_least=1, tags=['cluster:webs', 'varnish_name:default'])
 
 
 def test_inclusion_filter(aggregator, check, instance):
     instance['metrics_filter'] = ['SMA.*']
 
-    check.check(instance)
+    _check_repeat(check, instance)
     for mname in common.COMMON_METRICS:
         if 'SMA.' in mname:
-            aggregator.assert_metric(mname, count=1, tags=['cluster:webs', 'varnish_name:default'])
+            aggregator.assert_metric(mname, at_least=1, tags=['cluster:webs', 'varnish_name:default'])
         else:
             aggregator.assert_metric(mname, count=0, tags=['cluster:webs', 'varnish_name:default'])
 
 
 def test_exclusion_filter(aggregator, check, instance):
     instance['metrics_filter'] = ['^SMA.Transient.c_req']
 
-    check.check(instance)
+    _check_repeat(check, instance)
     for mname in common.COMMON_METRICS:
         if 'SMA.Transient.c_req' in mname:
             aggregator.assert_metric(mname, count=0, tags=['cluster:webs', 'varnish_name:default'])
         elif 'varnish.uptime' not in mname:
-            aggregator.assert_metric(mname, count=1, tags=['cluster:webs', 'varnish_name:default'])
+            aggregator.assert_metric(mname, at_least=1, tags=['cluster:webs', 'varnish_name:default'])
+
+
+def _check_repeat(check, instance):
+    # Ensure all metrics are generated
+    check.check(instance)
+    time.sleep(2)
+    check.check(instance)
```

Comment:
Seems a bit hacky :)

Would be better to add some wait conditions if the issue is that varnish is not ready for some reason: https://github.com/DataDog/integrations-core/blob/master/varnish/tests/conftest.py#L31-L32

Reply:
Varnish is ready is just some metrics seem not to be always emitted (It does not always fail with the same metric). I agree its hacky, another option is not to assert the metrics that are not always present

Reply:
> I agree its hacky, another option is not to assert the metrics that are not always present

That would be definitely worse.

The fix seems to work fine:

https://dev.azure.com/datadoghq/integrations-core/_build/results?buildId=1518&view=results